dask/widgets/__init__.py

```diff
@@ -0,0 +1,19 @@
+try:
+    from .widgets import FILTERS, TEMPLATE_PATHS, get_environment, get_template
+
+except ImportError as e:
+    msg = (
+        "Dask diagnostics requirements are not installed.\n\n"
+        "Please either conda or pip install as follows:\n\n"
+        "  conda install dask                     # either conda install\n"
+        '  python -m pip install "dask[diagnostics]" --upgrade  # or python -m pip install'
+    )
+    exception = e
```

Comment:
Nitpick: I'm not sure why this is needed. Is `except ImportError as exception:` a few lines above equivalent?

Reply:
It seems that the `e` gets dereferenced outside the `try` block despite it being used within the method.

MCRE

```python
try:
    raise Exception()
except Exception as e:
    def foo():
        raise Exception() from e

foo()
```

```python-traceback
Traceback (most recent call last):
  File "/tmp/test.py", line 7, in <module>
    foo()
  File "/tmp/test.py", line 5, in foo
    raise Exception() from e
NameError: name 'e' is not defined
```

So you need to assign it to a variable explicitly to maintain the reference.

```python
try:
    raise Exception()
except Exception as e:
    exception = e  # Make a reference here
    def foo():
        raise Exception() from exception

foo()
```

```python-traceback
Traceback (most recent call last):
  File "/tmp/test.py", line 2, in <module>
    raise Exception()
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/test.py", line 8, in <module>
    foo()
  File "/tmp/test.py", line 6, in foo
    raise Exception() from exception
Exception
```